src/snowflake/connector/cursor.py

```diff
@@ -704,6 +710,11 @@ def execute_async(self, *args, **kwargs):
         kwargs["_exec_async"] = True
         return self.execute(*args, **kwargs)
 
+    def describe(self, *args, **kwargs):
+        """Obtain the schema/description of the result without executing the query"""
```

Comment:
I'm not a fan of using `*args, **kwargs` on public APIs.
It would require a really good docstring that explains default values, description and type hints, but there's no automatic tool that can check this for us. So in general I don't allow them on public APIs, but since we want this function to automatically work with any arguments we add to `execute` let's do it with one addition.
Also return type-hint.
```suggestion
    def describe(self, *args, **kwargs) -> "SnowflakeCursor":
        """Obtain the schema/description of the result without executing the query.
        
        
        This function takes the same arguments as execute, please refer to that function
        for documentation.
        """
```